src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -353,103 +501,50 @@ void PosixEnginePollerManager::TriggerShutdown() {
 }
 
 std::shared_ptr<PosixEventEngine> PosixEventEngine::MakePosixEventEngine() {
-  // Constructor is private, can't use std::make_shared
-  return std::shared_ptr<PosixEventEngine>(new PosixEventEngine());
+  // Can't use make_shared as ctor is private
+  std::shared_ptr<PosixEventEngine> engine(new PosixEventEngine());
+  RegisterEventEngineForFork(engine);
+  return engine;
 }
 
-#ifdef GRPC_POSIX_SOCKET_TCP
-
-// The posix EventEngine returned by this method would have a shared ownership
-// of the poller and would not be in-charge of driving the poller by calling
-// its Work(..) method. Instead its up to the test to drive the poller. The
-// returned posix EventEngine will also not attempt to shutdown the poller
-// since it does not own it.
 std::shared_ptr<PosixEventEngine>
 PosixEventEngine::MakeTestOnlyPosixEventEngine(
     std::shared_ptr<grpc_event_engine::experimental::PosixEventPoller>
         test_only_poller) {
-  // Constructor is private, can't use std::make_shared
-  return std::shared_ptr<PosixEventEngine>(
+  std::shared_ptr<PosixEventEngine> engine(
       new PosixEventEngine(std::move(test_only_poller)));
-}
-#endif  // GRPC_POSIX_SOCKET_TCP
-
-PosixEnginePollerManager::~PosixEnginePollerManager() {
-  if (poller_ != nullptr) {
-    poller_->Shutdown();
-  }
+  RegisterEventEngineForFork(engine);
+  return engine;
 }
 
 PosixEventEngine::PosixEventEngine(std::shared_ptr<PosixEventPoller> poller)
     : grpc_core::KeepsGrpcInitialized(
           /*enabled=*/!grpc_core::IsPosixEeSkipGrpcInitEnabled()),
       connection_shards_(std::max(2 * gpr_cpu_num_cores(), 1u)),
       executor_(MakeThreadPool(grpc_core::Clamp(gpr_cpu_num_cores(), 4u, 16u))),
-      timer_manager_(std::make_shared<TimerManager>(executor_)) {
-  g_timer_fork_manager->RegisterForkable(
-      timer_manager_, TimerForkCallbackMethods::Prefork,
-      TimerForkCallbackMethods::PostforkParent,
-      TimerForkCallbackMethods::PostforkChild);
+      timer_manager_(executor_)
 #if GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
-  poller_manager_ = std::make_shared<PosixEnginePollerManager>(poller);
+      ,
```

Comment:
can the comma be on the same line as the assignment, or will clang-format complain?

Reply:
I used the approach you suggested in ares_resolver.cc - now looks cleaner to me.